src/test/java/com/tngtech/java/junit/dataprovider/internal/DataConverterTest.java

```diff
@@ -323,8 +315,7 @@ public void testConvertShouldReturnOneElementForStringArrayWithOneElementSplitBy
         List<Object[]> result = underTest.convert(data, false, parameterTypes, dataProvider);
 
         // Then:
-        assertThat(result).hasSize(1);
-        assertThat(result.get(0)).isEqualTo(new Object[] { "foo", true });
+        assertThat(result).containsExactly(new Object[] { "foo", true });
```

Comment:
I'm a fan of creating helper methods for tests. So if it's possible to come up with a descriptive method name I'm for adding it. Maybe something like `assertContainsOnlyObjectArrayWithValues(List, values...)`?

I agree that it would be better to write down the data explicitly in the test above.